plugins/destination/postgresql/client/migrate.go

```diff
@@ -48,165 +167,193 @@ func (c *Client) Migrate(ctx context.Context, tables schema.Tables) error {
 				return err
 			}
 		}
-		if err := c.Migrate(ctx, table.Relations); err != nil {
-			return err
-		}
+	}
+	conn, err := c.conn.Acquire(ctx)
+	if err != nil {
+		return fmt.Errorf("failed to acquire connection: %w", err)
+	}
+	defer conn.Release()
+	if err := conn.Conn().DeallocateAll(ctx); err != nil {
+		return fmt.Errorf("failed to deallocate all prepared statements: %w", err)
 	}
 	return nil
 }
 
-func (c *Client) isTableExistSQL(ctx context.Context, table string) (bool, error) {
-	var tableExist int
-	if err := c.conn.QueryRow(ctx, isTableExistSQL, table).Scan(&tableExist); err != nil {
-		return false, fmt.Errorf("failed to check if table %s exists: %w", table, err)
+func (c *Client) alterColumn(ctx context.Context, tableName string, column schema.Column) error {
+	switch c.pgType {
+	case pgTypeCockroachDB:
+		return c.alterColumnCockroachDB(ctx, tableName, column)
+	case pgTypePostgreSQL:
+		return c.alterColumnPg(ctx, tableName, column)
+	default:
+		return fmt.Errorf("unknown pg type: %v", c.pgType)
 	}
-	return tableExist == 1, nil
 }
 
-func (c *Client) getStalePks(pgPKs map[string]bool, table *schema.Table) []string {
-	stalePks := []string{}
-	if c.enabledPks() {
-		sortedPKs := maps.Keys(pgPKs)
-		sort.Strings(sortedPKs)
-		for _, pk := range sortedPKs {
-			stalePk := true
-			for _, col := range table.Columns {
-				if col.Name == pk && col.CreationOptions.PrimaryKey {
-					stalePk = false
-					break
-				}
-			}
-			if stalePk {
-				c.logger.Info().Str("table", table.Name).Str("column", pk).Msg("Column exists with primary key but is not in the schema")
-				stalePks = append(stalePks, pk)
-			}
+func (c *Client) alterColumnCockroachDB(ctx context.Context, tableName string, column schema.Column) error {
+	columnName := pgx.Identifier{column.Name}.Sanitize()
+	columnType := c.SchemaTypeToPg(column.Type)
+	// try alter column in place first
+	sql := "alter table " + tableName + " alter column " + columnName + " type " + columnType
+	if column.CreationOptions.NotNull {
+		sql += " set not null"
```

Comment:
```suggestion
		sql += ", ALTER " + columnName + " set not null"
```

Otherwise this always fail. Also Should we return early when it's successful so we don't need to drop/add it later one?